Allow empty lambda body

findepi opened the pull request.

electrum commented:
Nice, I didn't know they added that.

Note that there are actually a lot more instances of `-> { }` in the code base. We should probably either update all or none. However, since we're not enforcing they are empty, and IntelliJ will format down to one space, it's likely they will return.

findepi commented:
> Note that there are actually a lot more instances of `-> { }` in the code base.

Indeed there are many. Not sure how I did search showing up only those I fixed initially. I now fixed remaining ones.

> IntelliJ will format down to one space, it's likely they will return.

True, IntelliJ retains both `-> { }` and `-> {}` unchanged. Still, version without space (or newline) is more readable, so why not have it where possible now?

kokosing commented on src/checkstyle/checks.xml:
```diff
@@ -164,6 +164,7 @@
         <module name="WhitespaceAround">
             <property name="allowEmptyConstructors" value="true" />
             <property name="allowEmptyMethods" value="true" />
+            <property name="allowEmptyLambdas" value="true" />
```
Does it mean that now I can either use: `() -> { }` or `() -> {}` and checkstyle will be happy in both cases?

I would prefer to enforce one of them. 

Can we do the same with empty methods (constructors), I mean can enforce one way of formatting them: 

```
Constructror() {}
```
vs
```
Constructor() 
{
}
```

Maybe simple regex could be enough that disallows a white space character between `{}`?

findepi replied:
Yea, it is possible. I refrained from doing so, because currently the codebase uniformly uses `{\n}` for empty catch clauses (with single 1 exception). I didn't want to change that unilaterally, especially that it wouldn't necessarily improve readability.

OTOH, my main motivation was to reduce problems for developers. Lambdas are generally used in nested context, so making them more readable helps everyone. It's not obvious to me that being strict on empty constructors/methods is as helpful (it increases "oh Travis said i'm bad" ratio....)

electrum replied:
I'd personally prefer to enforce `{}` everywhere, as it's more consistent and signals that the block as intentionally left empty. But we should tackle that separately (assuming there is agreement for try/catch or other non-method blocks).

kokosing commented on presto-verifier/src/main/java/com/facebook/presto/verifier/Verifier.java:
```diff
@@ -168,7 +168,8 @@ public int run(List<QueryPair> queries)
                     ((Closeable) eventClient).close();
                 }
-                catch (IOException ignored) { }
+                catch (IOException ignored) {
```
why not:
```
catch (IOException) {}
```
?

findepi replied:
Because such is the prevailing style for empty catch bodies (every other place has new line).

kokosing commented on src/checkstyle/checks.xml:
```diff
@@ -35,6 +35,10 @@
     </module>
+    <module name="RegexpMultiline">
+        <property name="format" value="->\s*\{\s+\}" />
+        <property name="message" value="Whitespace inside empty lambda body" />
```
It does not necessarily mean that it is lambda.

kokosing replied:
Ah there is `->`, then ok. 

What about `\s*` is possible to have there multiple or none space between `->` and `{}`? Maybe you can replace that with simple ` ` (single space)?

findepi replied:
Multiple (or no) spaces between `->` and `{` don't make spaces inside
`{..}` more legal.
I didn't want to unnecessarily depend on other rules here

findepi commented:
Multiple (or no) spaces between `->` and `{` don't make spaces inside
`{..}` more legal.
I didn't want to unnecessarily depend on other rules here

findepi commented:
squashed & rebased. Can't make travis green since `master` isn't green. One configuration for PR build always fails (not the first, which runs checkstyle).
@kokosing @electrum

electrum commented:
I filed https://youtrack.jetbrains.com/issue/IDEA-178733 to have IntelliJ remove all spaces

findepi commented:
@electrum not sure what the conclusion is as far as this PR is concerned.

electrum commented:
Merged, thanks!
